app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/HedvigBottomSheet.kt

```diff
@@ -25,38 +56,111 @@ import eu.wewox.modalsheet.ModalSheet
 fun HedvigBottomSheet(
   isVisible: Boolean,
   onVisibleChange: (Boolean) -> Unit,
+  bottomButtonText: String,
   onSystemBack: (() -> Unit)? = { onVisibleChange(false) },
+  onBottomButtonClick: () -> Unit = { onVisibleChange(false) },
   sheetPadding: PaddingValues? = null,
   cancelable: Boolean = true,
   content: @Composable ColumnScope.() -> Unit,
 ) {
+  val scrollState = rememberScrollState()
+  var scrollDown by remember { mutableStateOf(false) }
+  LaunchedEffect(scrollDown) {
+    if (scrollDown) {
+      scrollState.animateScrollTo(scrollState.maxValue)
+    }
+  }
   val isImeVisible = WindowInsets.isImeVisible
   val defaultPadding = if (isImeVisible) {
     WindowInsets.ime.asPaddingValues()
   } else {
     WindowInsets.safeDrawing
-      .only(WindowInsetsSides.Bottom).asPaddingValues()
+      .only(WindowInsetsSides.Bottom + WindowInsetsSides.Top).asPaddingValues()
   }
   val finalSheetPadding = sheetPadding ?: defaultPadding
+
   ModalSheet(
     visible = isVisible,
     onVisibleChange = onVisibleChange,
     cancelable = cancelable,
-    content = content,
     scrimColor = bottomSheetColors.scrimColor,
     backgroundColor = bottomSheetColors.bottomSheetBackgroundColor,
     contentColor = bottomSheetColors.contentColor,
     sheetPadding = finalSheetPadding,
     onSystemBack = onSystemBack,
     shape = bottomSheetShape.shape,
+  ) {
+    Box {
+      Column(
+        modifier = Modifier.padding(horizontal = bottomSheetShape.contentHorizontalPadding)
+          .verticalScroll(scrollState),
+      ) {
+        Spacer(modifier = Modifier.height(8.dp))
+        LittleUpperChip()
+        Spacer(modifier = Modifier.height(20.dp))
+        content()
+        HedvigButton(
+          onClick = onBottomButtonClick,
+          text = bottomButtonText,
+          enabled = true,
+          buttonStyle = Ghost,
+          modifier = Modifier.fillMaxWidth(),
+        )
+        Spacer(modifier = Modifier.height(32.dp))
+      }
+      if (scrollState.canScrollForward && !scrollDown) {
+        HintArrowDown {
+          scrollDown = true
+        }
+      }
+    }
+  }
+}
+
+@Composable
+private fun ColumnScope.LittleUpperChip() {
```

Comment:
Same concept here. The CenterHorizontally part can come from the parent, and this composable can take in a `modifier` and not be an extension on `ColumnScope`.
The width/height etc can stay in here since this composable should just render the handle itself standalone without having to know how the parent will use it.